mantle/kola/harness.go

```diff
@@ -844,13 +849,34 @@ func metadataFromTestBinary(executable string) (*externalTestMeta, error) {
 	defer f.Close()
 	r := bufio.NewReader(io.LimitReader(f, 8192))
 	meta := &externalTestMeta{Exclusive: true}
+	inmeta := false    // true if we saw a ## kola: prefix after which we expect YAML
+	metadatayaml := "" // accumulated YAML metadata
 	for {
 		line, err := r.ReadString('\n')
 		if err == io.EOF {
 			break
 		} else if err != nil {
 			return nil, err
 		}
+		if strings.HasPrefix(line, InstalledTestMetaPrefixYaml) {
+			if inmeta {
+				return nil, fmt.Errorf("found multiple %s", InstalledTestMetaPrefixYaml)
+			}
+			inmeta = true
+			continue
+		}
+		if inmeta {
+			if !strings.HasPrefix(line, "## ") {
+				if err := yaml.Unmarshal([]byte(metadatayaml), &meta); err != nil {
+					return nil, err
+				}
+				break
+			} else {
+				metadatayaml += fmt.Sprintf("%s\n", line[3:])
+			}
+			continue
+		}
+		// Otherwise, we look for the legacy JSON metadata
 		if !strings.HasPrefix(line, InstalledTestMetaPrefix) {
 			continue
 		}
```

Comment:
optional: I think this code would read better now if it were also inside the `if`:

```suggestion
		// Otherwise, we look for the legacy JSON metadata
		if strings.HasPrefix(line, InstalledTestMetaPrefix) {
			buf := strings.TrimSpace(line[len(InstalledTestMetaPrefix):])
			...
			...
		}
```